### ISSUE TITLE
Warning: Accessing non-existent property of module exports inside circular

### ISSUE THREAD

**Issue by knoxcard2** [Unitech/pm2#4683]
After running this...
```
root@LAPTOP-21F611179:~/sites/buycanna.io# pm2 start app.js -i 4
```

I get this...
```
[PM2] Starting /root/sites/buycanna.io/app.js in cluster_mode (4 instances)
(node:3459) Warning: Accessing non-existent property 'cat' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:3459) Warning: Accessing non-existent property 'cd' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'chmod' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'cp' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'dirs' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'pushd' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'popd' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'echo' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'tempdir' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'pwd' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'exec' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'ls' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'find' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'grep' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'head' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'ln' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'mkdir' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'rm' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'mv' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'sed' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'set' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'sort' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'tail' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'test' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'to' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'toEnd' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'touch' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'uniq' of module exports inside circular dependency
(node:3459) Warning: Accessing non-existent property 'which' of module exports inside circular dependency
[PM2] Done.
```

**Comment by Unitech** [comment]
Working on it

**Comment by Unitech** [comment]
fixed on PM2 4.4.0
```
npm install pm2@latest -g
pm2 update
```

**Comment by ma6254** [comment]
Does pm2 support windows?

```powershell
PS C:\tools\pm2_config> pm2 update
Be sure to have the latest version by doing `npm install pm2@latest -g` before doing this procedure.
[PM2] [v] Modules Stopped
[PM2] Applying action deleteProcessId on app [all](ids: [ 0 ])
[PM2] [pm2-logrotate](0) ✓
[PM2] [v] All Applications Stopped
[PM2] [v] PM2 Daemon Stopped
[PM2] Spawning PM2 daemon with pm2_home=C:\Users\mjc\.pm2
(node:6844) Warning: Accessing non-existent property 'cat' of module exports inside circular dependency
(Use `node --trace-warnings ...` to show where the warning was created)
(node:6844) Warning: Accessing non-existent property 'cd' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'chmod' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'cp' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'dirs' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'pushd' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'popd' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'echo' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'tempdir' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'pwd' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'exec' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'ls' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'find' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'grep' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'head' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'ln' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'mkdir' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'rm' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'mv' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'sed' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'set' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'sort' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'tail' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'test' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'to' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'toEnd' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'touch' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'uniq' of module exports inside circular dependency
(node:6844) Warning: Accessing non-existent property 'which' of module exports inside circular dependency
[PM2] Restoring processes located in C:\Users\mjc\.pm2\dump.pm2
>>>>>>>>>> PM2 updated
[PM2][Module] Starting NPM module pm2-logrotate
[PM2][WARN] Applications pm2-logrotate not running, starting...
[PM2] App [pm2-logrotate] launched (1 instances)
node:internal/errors:541
      throw error;
      ^

TypeError [ERR_INVALID_ARG_TYPE]: The "data" argument must be of type string or an instance of Buffer, TypedArray, or DataView. Received type number (12008)
    at Object.writeFileSync (node:fs:2376:5)
    at InteractorDaemonizer.daemonize (C:\Users\mjc\AppData\Local\Yarn\Data\global\node_modules\@pm2\agent\src\InteractorClient.js:214:8)
    at C:\Users\mjc\AppData\Local\Yarn\Data\global\node_modules\@pm2\agent\src\InteractorClient.js:298:14
    at ReqSocket.<anonymous> (C:\Users\mjc\AppData\Local\Yarn\Data\global\node_modules\@pm2\agent\src\InteractorClient.js:44:14)
    at Object.onceWrapper (node:events:633:28)
    at ReqSocket.emit (node:events:519:28)
    at Timeout._onTimeout (C:\Users\mjc\AppData\Local\Yarn\Data\global\node_modules\pm2-axon\lib\sockets\sock.js:279:12)
    at listOnTimeout (node:internal/timers:581:17)
    at process.processTimers (node:internal/timers:519:7) {
  code: 'ERR_INVALID_ARG_TYPE'
}

Node.js v20.16.0
```
